lib/header.ml

```diff
@@ -16,16 +16,30 @@
  *
   }}}*)
 
+exception Found_int of int
+
+let compare_caseless x y =
+  if String.(length x <> length y)
+  then String.compare x y
+  else
+    try
+      for i = 0 to String.length x - 1 do
+        if Char.(lowercase x.[i] <> lowercase y.[i]) then
+          raise (Found_int (Char.compare x.[i] y.[i]))
+      done;
+      0
+    with Found_int i -> i
```

Comment:
Sorry, this bit scrolled out of scope on my phone :) Ignore the previous "dont leak exception comment"